src/main/java/mil/dds/anet/utils/Utils.java

```diff
@@ -323,27 +323,79 @@ public static boolean isEmptyOrNull(Collection<?> c) {
    * @param whitelistDomainNames The list of whitelisted domain names (wildcards allowed)
    * @return Whether the email is whitelisted
    */
-  public static boolean isEmailWhitelisted(String email, List<String> whitelistDomainNames) {
-    if (isEmptyOrNull(email) || isEmptyOrNull(whitelistDomainNames)) {
+  public static boolean isEmailWhitelisted(final String email,
+      final List<String> whitelistDomainNames) {
+    try {
+      return isLogonDomainInList(email, whitelistDomainNames);
+    } catch (Exception e) {
+      logger.error("Failed to process email: ", email);
       return false;
     }
+  }
+
+  /**
+   * Checks whether an domain user name is allowed according to a list of whitelisted domains.
+   * 
+   * More info: https://docs.microsoft.com/en-us/windows/win32/secauthn/user-name-formats
+   * 
+   * @param email The domain user name to check
+   * @param whitelistDomainNames The list of ignaored domain user names (wildcards allowed)
+   * @return Whether the domain user name is ignored
+   */
+  public static boolean isDomainUserNameIgnored(final String domainUserName,
+      final List<String> ignoredDomainNames) {
+    try {
+      return isLogonDomainInList(domainUserName, ignoredDomainNames);
+    } catch (Exception e) {
+      logger.error("Failed to process domain user name: ", domainUserName);
+      return true;
+    }
+  }
+
+  private static boolean isLogonDomainInList(final String logon, final List<String> list)
+      throws Exception {
```

Comment:
Please use something more specific than `Exception` (perhaps create a custom exception).

Reply:
Good point. I would rather use the standard Java Exceptions whether possible (and appropiate). In this case I was doubting between the more general RuntimeException and IllegalArgumentException, but I chose the latter because the API states that: Thrown to indicate that a method has been passed an illegal or inappropriate argument.